src/components/Navigation.vue

```diff
@@ -129,6 +132,14 @@ export default {
         this.subselectionType = val[0];
       }
     },
+    currentComponent() {
+      if (!this.currentComponent) { return; }
+      if (this.modeTab !== 'components') { return; }
+
+      this.subselectionType = this.currentComponent.type === 'window_definitions' ? 'windows' :
```

Comment:
This might be a bit confusing for future readers (but is totally a matter of personal preference)